opentelemetry/src/sdk/trace/tracer.rs

```diff
@@ -253,18 +254,31 @@ impl crate::trace::Tracer for Tracer {
             span_trace_state = trace_state;
             attribute_options.append(&mut extra_attrs);
             let mut attributes =
-                EvictedHashMap::new(config.max_attributes_per_span, attribute_options.len());
+                EvictedHashMap::new(span_limit.max_attributes_per_span, attribute_options.len());
             for attribute in attribute_options {
                 attributes.insert(attribute);
             }
-            let mut links = EvictedQueue::new(config.max_links_per_span);
+            let mut links = EvictedQueue::new(span_limit.max_links_per_span);
             if let Some(link_options) = &mut link_options {
+                for link in link_options.iter_mut() {
```

Comment:
Do we still need to do this? going through `append_vec` below will call [`push_back`](https://github.com/open-telemetry/opentelemetry-rust/blob/main/opentelemetry/src/sdk/trace/evicted_queue.rs#L32-L35) which should already handle the limits configured above

Reply:
The `append_vec`  only enforces the limit on the number of links. Not the number of attributes of a link.

Reply:
Oh sorry I see what's happening nvm

Reply:
It may make sense to redesign how we create a link to enforce the limit when we creating one :thinking: in the future. Gonna have to use `EvictedQueue` to store the attributes too

Reply:
Yeah can rethink he we store dropped count and others in a follow up PR perhaps

Reply:
My current plan is to add fields in `SpanLimits` to allow us to store the dropped number in it. 
Also need to print the warning message when the dropped number reaches some threshold. Cannot really warn whenever we drop because the spec says don't be too noisy on this issue.